sql/src/main/java/io/crate/lucene/AbstractAnyQuery.java

```diff
@@ -44,16 +45,21 @@ public Query apply(Function function, LuceneQueryBuilder.Context context) throws
         Symbol collectionSymbol = function.arguments().get(1);
         Preconditions.checkArgument(DataTypes.isCollectionType(collectionSymbol.valueType()),
             "invalid argument for ANY expression");
+
+        if (DataTypes.isCollectionType(left.valueType())) {
+            throw new UnsupportedFeatureException(
```

Comment:
I find this confusing. I thought [1, 2, 3] = any (...) is something we do support. Am I missing something?

Reply:
`AnyEq` no longer uses this base class, and the other operations `AnyGt`, `AnyLt`, etc. are still not supported.

I see how this can be a bit confusing - any suggestions how to clear things up?

Reply:
Ah, I commented before scrolling. I guess it's fine like this as it applies to all current implementors (I'd otherwise delegate to the implementors to signal this with an abstract `isLeftCollectionSupported` but at it currently stands it's an overkill I think.)